Assets/_BuildingBlocks/DataTypeAdapters/DataTypeChain.cs

```diff
@@ -93,10 +97,23 @@ private IEnumerator DownloadAndCheckSupport(string url)
         private IEnumerator DownloadDataToLocalCache(LocalFile urlAndData)
         {
             var futureFileInfo = Resource<FileInfo>.At(urlAndData.SourceUrl).Value;
-            
+
             // We want to use and manipulate urlAndData, so we 'curry' it by wrapping a method call in a lambda 
-            futureFileInfo.Then(info => DownloadSucceeded(urlAndData, info));
-            futureFileInfo.Catch(error => DownloadFailed(urlAndData, error));
+            futureFileInfo.Then(info =>
+            {                
+                DownloadSucceeded(urlAndData, info);
+            });        
+            futureFileInfo.Catch(error =>
+            {
+                if (error is AuthenticationError)
+                {
+                    OnAuthenticationFailed?.Invoke(urlAndData.SourceUrl);
+                }
+                else
```

Comment:
Else is niet nodig als je in de if hierboven een return gebruikt - early return concept. Scheelt nesting en maakt je code leesbaarder